klog.go

```diff
@@ -918,6 +918,13 @@ func LogToStderr(stderr bool) {
 // output writes the data to the log files and releases the buffer.
 func (l *loggingT) output(s severity, log *logr.Logger, buf *buffer, depth int, file string, line int, alsoToStderr bool) {
 	l.mu.Lock()
+	defer func() {
+		if err := recover(); err != nil {
+			l.mu.Unlock()                                                // Make sure to do `Unlock` if panic occurs below (eg: #L938 Error)
+			os.Stderr.Write([]byte("panic observed in loggingT.output")) // Make sure the message appears somewhere.
```

Comment:
If the logger panicked, why should that panic be recovered instead of aborting the process?

Reply:
Got it, updated with aborting the process.

Reply:
I would prefer to not recover the panic at all. If logging panics, then let the caller deal with it.

The logic here with "if there's a panic, we probably didn't unlock" seems fragile. I would prefer a solution with a `isLocked := true` that gets set to false in those places where this functions must unlock and then have
```
defer func() {
    if isLocked {
        l.mu.Unlock()
    }
}()
```

The normal unlock at the end of this function then can be removed.

One question, though: does this have a performance impact?
Okay, two questions: does it matter?

Reply:
1. Updated defer func with  `isLocked` bool value;
2. Added benchmark test for lock with/without defer, the result is:
go version go1.16 darwin/amd64
`go test -bench=.`
==========================
goos: darwin
goarch: amd64
pkg: k8s.io/klog/v2
cpu: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
...
BenchmarkWithoutDeferUnLock-12          95588708                12.46 ns/op
BenchmarkWithDeferUnLock-12             92082324                12.87 ns/op
PASS
ok      k8s.io/klog/v2  10.800s

From the benchmark result, we can see the defer unlock just has a little performance impact.
So it is safe to use defer, PTAL thanks.